app/models/manager_refresh/inventory_collection.rb

```diff
@@ -368,14 +370,18 @@ class InventoryCollection
     # @param use_ar_object [Boolean] True or False. Whether we need to initialize AR object as part of the saving
     #        it's needed if the model have special setters, serialize of columns, etc. This setting is relevant only
     #        for the batch saver strategy.
+    # @param batch_extra_attributes [Array] Array of symbols marking which extra attributes we want to store into the
+    #        db. These extra attributes might be a product of :use_ar_object assignment and we need to specify them
+    #        manually, if we want to use a batch saving strategy and we have models that populate attributes as a side
+    #        effect.
     def initialize(model_class: nil, manager_ref: nil, association: nil, parent: nil, strategy: nil, saved: nil,
                    custom_save_block: nil, delete_method: nil, data_index: nil, data: nil, dependency_attributes: nil,
                    attributes_blacklist: nil, attributes_whitelist: nil, complete: nil, update_only: nil,
                    check_changed: nil, custom_manager_uuid: nil, custom_db_finder: nil, arel: nil, builder_params: {},
                    inventory_object_attributes: nil, unique_index_columns: nil, name: nil, saver_strategy: nil,
                    parent_inventory_collections: nil, manager_uuids: [], all_manager_uuids: nil, targeted_arel: nil,
                    targeted: nil, manager_ref_allowed_nil: nil, secondary_refs: {}, use_ar_object: nil,
-                   custom_reconnect_block: nil)
+                   custom_reconnect_block: nil, batch_extra_attributes: [])
```

Comment:
Seems like there are a ridiculous number of parameters to this `initialize` method.  Sandy Metz [recommends](https://robots.thoughtbot.com/sandi-metz-rules-for-developers) no more than 4.  Feels like this is a bad pattern that can be refactored in future PRs.  /cc @agrare